### PR TITLE
Allow shooting out lights in the lab.

### PR DESCRIPTION
**Pull request by OzoneH3**
```SUMMARY: Features "Allow shooting out lights in the lab."```

#### Purpose of change
https://www.reddit.com/r/cataclysmdda/comments/dz1e8x/as_someone_who_spent_a_lot_of_time_in_labs/

#### Describe the solution
Added entry for ```t_thconc_floor_olight``` terrain to huge if else block.

#### Describe alternatives you've considered
Moving that whole if else mess into json. Was too lazy to do so.

#### Testing
Shot at light. Light dead.

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -3792,6 +3792,12 @@ void map::shoot( const tripoint &p, projectile &proj, const bool hit_items )
         } else {
             dam = 0;
         }
+    } else if( terrain == t_thconc_floor_olight ) {
+        break_glass( p, 16 );
+        ter_set( p, t_thconc_floor );
+        spawn_item( p, "glass_shard", rng( 8, 16 ) );
+        dam = 0; //Prevent damaging additional items, sice we shot at the ceiling.
```
Comment (Night-Pryanik) [inline src/map.cpp]:
```suggestion
        dam = 0; //Prevent damaging additional items, since we shot at the ceiling.
```

**Comment by Night-Pryanik** [comment]
Maybe add some chance to hit instead of guaranteed?

**Comment by OzoneH3** [comment]
Gave it a 1 in 3 chance. At this point in the code there is no information about accuracy/distance/etc. available.

**Comment by agaume** [comment]
My hero <3

**Comment by BreBat** [comment]
This is allowing "overhead lights" to block shots intended for targets on the ground

**Comment by ZhilkinSerg** [comment]
> This is allowing "overhead lights" to block shots intended for targets on the ground

No?

**Comment by OzoneH3** [comment]
As far as I remember this only gets called when shooting at an empty tile or the monster was missed with the shot.

**Comment by BreBat** [comment]
Ah, apologies if I was mistaken. I just noticed this behavior for the first time and assumed it had something to do with this pull. 

image

**Comment by ZhilkinSerg** [comment]
I guess you missed your target, so you hit a terrain instead. It could happen with other terrains too.

**Pull request merged**